solnlib/bulletin_rest_client.py

```diff
@@ -0,0 +1,140 @@
+#
+# Copyright 2024 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from solnlib import splunk_rest_client as rest_client
+from splunklib import binding
+from typing import Optional, List
+import json
+
+__all__ = ["BulletinRestClient"]
+
+
+class BulletinRestClient:
+    """REST client for handling Bulletin messages."""
+
+    MESSAGES_ENDPOINT = "/services/messages"
+
+    headers = [("Content-Type", "application/json")]
+
+    class Severity:
+        INFO = "info"
+        WARNING = "warn"
+        ERROR = "error"
+
+    def __init__(
+        self,
+        message_name: str,
+        session_key: str,
+        **context: dict,
+    ):
+        """Initializes BulletinRestClient.
+
+            One instance is responsible for handling one, particular message.
+            If you need to add another message to bulletin create another instance
+            with a different 'message_name'
+            e.g.
+            msg_1 = BulletinRestClient("message_1", "<some session key>")
+            msg_2 = BulletinRestClient("message_2", "<some session key>")
+
+        Arguments:
+            message_name: Name of the message in the splunk's bulletin.
+            session_key: Splunk access token.
+            context: Other configurations for Splunk rest client.
+        """
+
+        self.message_name = message_name
+        self.session_key = session_key
+
+        self._rest_client = rest_client.SplunkRestClient(
+            self.session_key, app="-", **context
```

Comment:
Should we provide a particular here `app="-"`? Is there any consequences of using `-`?

Reply:
`app` is a mandatory argument for `SplunkRestClient` but it is used to call `client.Service` where is an optional parameter. 
`:param `app`: The app context of the namespace (optional; use "-" for wildcard).`
I used '-' as it was done in some other modules from solnlib, but maybe it's better to let the developer decide what he wants to do with it. Changed.